src/payment/strategies/affirm/affirm-payment-strategy.spec.ts

```diff
@@ -0,0 +1,254 @@
+            paymentActionCreator,
+            remoteCheckoutActionCreator
+        );
+
+        paymentMethod = getAffirm();
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethod.id,
+                gatewayId: paymentMethod.gateway,
+            },
+        });
+
+        initializePaymentAction = of(createAction(RemoteCheckoutActionType.InitializeRemotePaymentRequested));
+        loadRemoteSettingsAction = of(createAction(
+            RemoteCheckoutActionType.LoadRemoteSettingsSucceeded,
+            { useStoreCredit: false },
+            { methodId: paymentMethod.id }
+        ));
+        submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+        submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+
+        jest.spyOn(store, 'dispatch');
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(remoteCheckoutActionCreator, 'initializePayment')
+            .mockReturnValue(initializePaymentAction);
+
+        jest.spyOn(remoteCheckoutActionCreator, 'loadSettings')
+            .mockReturnValue(loadRemoteSettingsAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethod.id,
+                gatewayId: paymentMethod.gateway,
+            },
+        });
+    });
+
+    describe('#initialize()', () => {
+        it('throws an exception if payment method cannot be found', () => {
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(undefined);
+            const error = 'Unable to proceed because payment method data is unavailable or not properly configured.';
+            expect(() => strategy.initialize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway })).toThrowError(error);
+        });
+    });
+
+    describe('#execute()', () => {
+        const successHandler = jest.fn();
+
+        beforeEach(async () => {
+            await strategy.initialize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway });
+
+            strategy.execute(payload).then(successHandler);
+
+            await new Promise(resolve => process.nextTick(resolve));
+        });
+
+        it('notifies store credit usage to remote checkout service', () => {
+            expect(remoteCheckoutActionCreator.initializePayment).toHaveBeenCalledWith( paymentMethod.id, { useStoreCredit: false });
+            expect(store.dispatch).toHaveBeenCalledWith(initializePaymentAction);
+        });
+
+        it('does not resolve if execution is successful', () => {
+            expect(successHandler).not.toHaveBeenCalled();
```

Comment:
`successHandler` also doesn't get called when there's a failure. So I'm not sure this is the correct verification.